assets/js/modules/tagmanager/components/setup/SetupForm.js

```diff
@@ -16,64 +16,105 @@
 import {
 	AccountSelect,
 	AMPContainerSelect,
-	FormInstructions,
 	WebContainerSelect,
 } from '../common';
 import StoreErrorNotice from '../../../../components/StoreErrorNotice';
+import SetupFormInstructions from './SetupFormInstructions';
+import SetupFormSubmitButtons from './SetupFormSubmitButtons';
 const { useSelect, useDispatch } = Data;
 
-export default function SetupForm( { finishSetup } ) {
-	const canSubmitChanges = useSelect( ( select ) => select( STORE_NAME ).canSubmitChanges() );
+export default function SetupForm( { finishSetup, setIsNavigating } ) {
+	const gtmAnalyticsPropertyID = useSelect( ( select ) => select( STORE_NAME ).getSingleAnalyticsPropertyID() );
+	const analyticsModuleActive = useSelect( ( select ) => select( CORE_MODULES ).isModuleActive( 'analytics' ) );
 	const hasEditScope = useSelect( ( select ) => select( CORE_USER ).hasScope( EDIT_SCOPE ) );
 	const autoSubmit = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_SETUP, 'autoSubmit' ) );
+	const submitMode = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_SETUP, 'submitMode' ) );
+	const analyticsModuleReauthURL = useSelect( ( select ) => select( MODULES_ANALYTICS ).getAdminReauthURL() );
 
 	const { setValues } = useDispatch( CORE_FORMS );
-	const { submitChanges } = useDispatch( STORE_NAME );
-	const submitForm = useCallback( async ( event ) => {
-		event.preventDefault();
+	const { activateModule } = useDispatch( CORE_MODULES );
+	const { submitChanges, receiveError } = useDispatch( STORE_NAME );
+	const dispatchAnalytics = useDispatch( MODULES_ANALYTICS );
+	const submitForm = useCallback( async ( { submitMode } ) => { // eslint-disable-line no-shadow
+		// We'll use form state to persist the chosen submit choice
+		// in order to preserve support for auto-submit.
+		setValues( FORM_SETUP, { submitMode } );
+
 		const { error } = await submitChanges();
+
 		if ( isPermissionScopeError( error ) ) {
 			setValues( FORM_SETUP, { autoSubmit: true } );
-		}
-		if ( ! error ) {
+		} else if ( ! error ) {
 			setValues( FORM_SETUP, { autoSubmit: false } );
+			// If the property ID is set in GTM and Analytics is active,
+			// we disable the snippet output via Analyics to prevent duplicate measurement.
+			if ( gtmAnalyticsPropertyID && analyticsModuleActive ) {
+				dispatchAnalytics.setUseSnippet( false );
+				const saveAnalyticsSettings = await dispatchAnalytics.saveSettings();
+				if ( saveAnalyticsSettings.error ) {
+					receiveError( saveAnalyticsSettings.error );
+					return;
+				}
+			}
+			// If submitting with Analytics setup, and Analytics is not active,
+			// activate it, and navigate to its reauth/setup URL to proceed with its setup.
+			if ( submitMode === 'with_analytics_setup' && ! analyticsModuleActive ) {
+				setIsNavigating( true );
+				const activateAnalytics = await activateModule( 'analytics' );
+				if ( activateAnalytics.error ) {
+					receiveError( activateAnalytics.error );
+					return;
```

Comment:
I think we'll need to call `setIsNavigating( false )` here to avoid a continuing progress bar in case of this error?

Reply:
I've refactored the function here to use a try/catch so we only need to set it back to `false` if any of the requests fail 🙂